alacritty_terminal/src/term/mod.rs

```diff
@@ -687,6 +687,83 @@ impl VisualBell {
     }
 }
 
+/// LineDamage describes the current damage of a line, represented as two
+/// column coordinates describing the edges of the damage. Damage is valid if
+/// left <= right.
+#[derive(Clone)]
+pub struct LineDamage {
+    pub left: Column,
+    pub right: Column,
+    pub line: Line,
+}
+
+impl LineDamage {
+    /// Create a new, undamaged LineDamage. where left=num_cols+1 and right=0.
+    /// This ensures that later update with min/max of any valid column
+    /// automatically creates a valid line damage without needing any
+    /// conditionals.
+    fn undamaged(line: Line, num_cols: Column) -> LineDamage {
+        LineDamage { left: num_cols + 1, right: Column(0), line }
```

Comment:
it should be `- 1` and not `+ 1`, otherwise this column will be out of grid.

Reply:
It was initially intentional, but it doesn't matter as long as left is at least the rightmost extreme, right is at least the leftmost extreme. I'll change it to make it less confusing.

Reply:
I do understand that it doesn't matter here from the algo side, but lets just have kind of correct numbers.